File: Arrays/MaximumDifference/maximumDifference.cpp
#include<iostream>
using namespace std;

int maxDiffernce(int arr[], int n)
{
    int res = arr[1]-arr[0];
    int mini = arr[0];
    for(int i = 0;i<n; i++){
        res = max(res, arr[i]-mini);
        mini = min(mini, arr[i]);
    }
    return res;
}

int main()
{
    int arr[] = {2,3,10,6,4,8,1};
    int n = sizeof(arr)/sizeof(arr[0]);
    cout<<maxDiffernce(arr,n);
    return 0;
}

File: Arrays/MaximumDifference/maximumDifference.java
package Arrays.MaximumDifference;

public class maximumDifference {
    
    public static Integer maxDifference(Integer arr[]){
        Integer res = 0;
        res = arr[1]-arr[0];
        Integer min = arr[0];
        for(int i = 1;i<arr.length;i++){
            res = Math.max(res, arr[i]-min);
            min = Math.min(min, arr[i]);
        } 
        return res;
    }

    public static void main(String args[]){
        Integer arr[] = {2,3,10,6,4,8,1};
        System.out.println(maxDifference(arr));
    }
}
